### PR TITLE
[nrf52840] hal: Fixes inaccurate system time for micros()

### PR DESCRIPTION
**Pull request by YutingYou**
### Problem

micros() is not accurate in RC27, because we change the clock source from `DWT->CYCCNT` to `RTC2`, the time error become: 1/32.768KHz = 30.5us

### Solution

Using RTC as the main source but artificially decrease the error using DWT->CYCCNT

### Steps to Test

Flash below example

### Example App

```c
void setup() {
    Serial1.begin(115200);
}

void loop() {

    SINGLE_THREADED_BLOCK() {
        uint32_t delta = micros();
        delayMicroseconds(93);
        delta = micros() - delta;
        Serial1.printlnf("delay 93us, result: %dus", delta);

        delta = micros();
        delayMicroseconds(96);
        delta = micros() - delta;
        Serial1.printlnf("delay 96us, result: %dus", delta);

        delta = micros();
        delayMicroseconds(99);
        delta = micros() - delta;
        Serial1.printlnf("delay 99us, result: %dus \r\n", delta);
    }
    delay(1000);
}
```
#### before
```
delay 93us, result: 92us
delay 96us, result: 91us
delay 99us, result: 92us

delay 93us, result: 92us
delay 96us, result: 92us
delay 99us, result: 122us

delay 93us, result: 91us
delay 96us, result: 92us
delay 99us, result: 91us

delay 93us, result: 91us
delay 96us, result: 92us
delay 99us, result: 122us
```
#### after
```
delay 93us, result: 100us
delay 96us, result: 102us
delay 99us, result: 106us

delay 93us, result: 99us
delay 96us, result: 103us
delay 99us, result: 106us

delay 93us, result: 100us
delay 96us, result: 102us
delay 99us, result: 105us

delay 93us, result: 99us
delay 96us, result: 102us
delay 99us, result: 105us
```
### References

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: hal/src/nRF52840/openthread/alarm.c [hunk 1]
```diff
@@ -279,7 +282,7 @@ static void GetOffsetAndCounter(uint32_t *aOffset, uint32_t *aCounter)
 
 static uint64_t GetTime(uint32_t aOffset, uint32_t aCounter, AlarmIndex aIndex)
 {
-    uint64_t result = (uint64_t)aOffset * US_PER_OVERFLOW + TicksToTime(aCounter, kUsTimer);
+    uint64_t result = (uint64_t)aOffset * US_PER_OVERFLOW + (DWT->CYCCNT - sSystemMillisClock) / SYSTEM_US_TICKS;
```
Comment (avtolstoy) [inline hal/src/nRF52840/openthread/alarm.c]: I am a bit on the fence about this. I'd rather keep OT-specific functionality untouched and use `DWT` for higher precision only for our `hal_timer_micros()`.

File: hal/src/nRF52840/openthread/alarm.c [hunk 2]
```diff
@@ -104,6 +106,7 @@ static volatile uint8_t  sMutex;           ///< Mutex for write access to @ref s
 static volatile uint64_t sTimeOffset = 0;  ///< Time overflowCounter to keep track of current time (in millisecond).
 static volatile bool     sEventPending;    ///< Timer fired and upper layer should be notified.
 static AlarmData         sTimerData[kNumTimers]; ///< Data of the timers.
+static volatile uint32_t sSystemMillisClock;     ///< Use DWT->CYCCNT to make microseconds accurate
```
Comment (avtolstoy) [inline hal/src/nRF52840/openthread/alarm.c]: Should this be `sSystemTicks`?

File: hal/src/nRF52840/openthread/alarm.c (hunk 1 above)
Comment (YutingYou) [reply hal/src/nRF52840/openthread/alarm.c]: Sounds good! Fixed.

File: hal/src/nRF52840/openthread/alarm.c (hunk 2 above)
Comment (YutingYou) [reply hal/src/nRF52840/openthread/alarm.c]: Renamed.

**Comment by YutingYou** [comment]
Close this PR because it doesn't fix the inaccurate system time, we'll go back to using Systick + DWT.

**Pull request closed without merging**